src/main/java/org/wso2/carbon/inbound/salesforce/poll/EmpConnector.java

```diff
@@ -0,0 +1,394 @@
+
+        this.bearerTokenProvider = bearerTokenProvider;
+    }
+
+    /**
+     * Subscribe to a topic, receiving events after the replayFrom position
+     *
+     * @param topic      - the topic to subscribe to
+     * @param replayFrom - the replayFrom position in the event stream
+     * @param consumer   - the consumer of the events
+     * @return a Future returning the Subscription - on completion returns a Subscription or throws a CannotSubscribe
+     * exception
+     */
+    public Future<TopicSubscription> subscribe(String topic, long replayFrom, Consumer<Map<String, Object>> consumer) {
+
+        if (!running.get()) {
+            throw new IllegalStateException(String.format("Connector[%s} has not been started", parameters.endpoint()));
+        }
+        if (replay.putIfAbsent(topic, replayFrom) != null) {
+            throw new IllegalStateException(
+                    String.format("Already subscribed to %s [%s]", topic, parameters.endpoint()));
+        }
+        SubscriptionImpl subscription = new SubscriptionImpl(topic, consumer);
+        return subscription.subscribe();
+    }
+
+    /**
+     * Subscribe to a topic, receiving events from the earliest event position in the stream
+     *
+     * @param topic    - the topic to subscribe to
+     * @param consumer - the consumer of the events
+     * @return a Future returning the Subscription - on completion returns a Subscription or throws a CannotSubscribe
+     * exception
+     */
+    public Future<TopicSubscription> subscribeEarliest(String topic, Consumer<Map<String, Object>> consumer) {
+
+        return subscribe(topic, REPLAY_FROM_EARLIEST, consumer);
+    }
+
+    /**
+     * Subscribe to a topic, receiving events from the latest event position in the stream
+     *
+     * @param topic    - the topic to subscribe to
+     * @param consumer - the consumer of the events
+     * @return a Future returning the Subscription - on completion returns a Subscription or throws a CannotSubscribe
+     * exception
+     */
+    public Future<TopicSubscription> subscribeTip(String topic, Consumer<Map<String, Object>> consumer) {
+
+        return subscribe(topic, REPLAY_FROM_TIP, consumer);
+    }
+
+    public EmpConnector addListener(String channel, ClientSessionChannel.MessageListener messageListener) {
+
+        listenerInfos.add(new MessageListenerInfo(channel, messageListener));
+        return this;
+    }
+
+    public boolean isConnected() {
+
+        return client != null && client.isConnected();
+    }
+
+    public boolean isDisconnected() {
+
+        return client == null || client.isDisconnected();
+    }
+
+    public boolean isHandshook() {
+
+        return client != null && client.isHandshook();
+    }
+
+    public long getLastReplayId(String topic) {
+
+        return replay.get(topic);
+    }
+
+    private Future<Boolean> connect() {
+
+        LOG.info("Connector connecting");

```

Comment:
Did you put this log in this method?. If then, you didn't initialize the logger

Reply:
log moved to the appropriate place and fixed